fuel/transformers/image.py

```diff
@@ -0,0 +1,263 @@
+    return iterables containing such objects. In the case of an
+    iterable, a list of loaded images is returned.
+
+    """
+    def __init__(self, data_stream, color_mode='RGB', **kwargs):
+        kwargs.setdefault('produces_examples', data_stream.produces_examples)
+        # Acrobatics currently required to correctly set axis labels.
+        which_sources = kwargs.get('which_sources', data_stream.sources)
+        axis_labels = self._make_axis_labels(data_stream, which_sources,
+                                             kwargs['produces_examples'])
+        kwargs.setdefault('axis_labels', axis_labels)
+        super(ImagesFromBytes, self).__init__(data_stream, **kwargs)
+        self.color_mode = color_mode
+
+    def transform_source_example(self, example, source_name):
+        if PY3:
+            bytes_type = bytes
+        else:
+            bytes_type = str
+        if not isinstance(example, bytes_type):
+            raise TypeError("expected {} object".format(bytes_type.__name__))
+        pil_image = Image.open(BytesIO(example))
+        if self.color_mode is not None:
+            pil_image = pil_image.convert(self.color_mode)
+        image = numpy.array(pil_image)
+        if image.ndim == 3:
+            # Transpose to `(channels, height, width)` layout.
+            return image.transpose(2, 0, 1)
+        elif image.ndim == 2:
+            # Add a channels axis of length 1.
+            image = image[numpy.newaxis]
+        else:
+            raise ValueError('unexpected number of axes')
+        return image
+
+    def transform_source_batch(self, batch, source_name):
+        return [self.transform_source_example(im, source_name) for im in batch]
+
+    def _make_axis_labels(self, data_stream, which_sources, produces_examples):
+        # This is ugly and probably deserves a refactoring of how we handle
+        # axis labels. It would be simpler to use memoized read-only
+        # properties, but the AbstractDataStream constructor tries to set
+        # self.axis_labels currently. We can't use self.which_sources or
+        # self.produces_examples here, because this *computes* things that
+        # need to be passed into the superclass constructor, necessarily
+        # meaning that the superclass constructor hasn't been called.
+        # Cooperative inheritance is hard, etc.
+        labels = {}
+        for source in data_stream.sources:
+            if source in which_sources:
+                if produces_examples:
+                    labels[source] = ('channel', 'height', 'width')
+                else:
+                    labels[source] = ('batch', 'channel', 'height', 'width')
+            else:
+                labels[source] = (data_stream.axis_labels[source]
+                                  if source in data_stream.axis_labels
+                                  else None)
+        return labels
+
+
+class MinimumImageDimensions(SourcewiseTransformer, ExpectsAxisLabels):
```

Comment:
For now, this transformer exists mainly to make sure that your images are big enough to sample a certain given window size. Other transformers implementing the "scale the shorter side to a particular pixel length" will be forthcoming over the next little while, but for now this is sufficient to guarantee the random crop transformer doesn't choke.